test/index.test.js

```diff
@@ -57,4 +59,306 @@ describe('react-ga', function() {
+      getGaCalls().should.eql([ [ 'create', 'foo', 'auto' ],
+                                [ 'send', { eventAction: 'Send Test',
+                                            eventCategory: 'Test',
+                                            nonInteraction: true,
+                                            hitType: 'event'
+                                          }]
+                                ]);
+    });
+
+    it('should reject a non-boolean nonInteraction value', function() {
+      ga.initialize('foo');
+      ga.event( { category: 'Test', action: 'Send Test', nonInteraction: 'yeahsure' } );
+      console.warn.args.should.eql([[
+        '[react-ga]', '`args.nonInteraction` must be a boolean.'
+      ]]);
+      getGaCalls().should.eql([ [ 'create', 'foo', 'auto' ],
+                                [ 'send', { eventAction: 'Send Test',
+                                            eventCategory: 'Test',
+                                            hitType: 'event'
+                                          }]
+                                ]);
+    });
+  });
+
+  /**
+   * outboundLink()
+   */
+
+  describe('outboundLink()', function() {
+
+    it('should record an outboundLink event', function() {
+
+      ga.initialize('foo');
+      ga.outboundLink( { label: 'Test Click' }, function () {
+      } );
+
+      getGaCalls().should.eql([ [ 'create', 'foo', 'auto' ],
+                                [ 'send', { eventAction: 'Click',
+                                            eventCategory: 'Outbound',
+                                            eventLabel: 'Test Click',
+                                            hitCallback: undefined,
+                                            hitType: 'event'
+                                          }]
+                                ]);
+
+    });
+
+    it('should warn if all args are missing', function() {
+      ga.initialize('foo');
+      ga.outboundLink();
+      console.warn.args.should.eql([[
+        '[react-ga]', 'hitCallback function is required'
+      ]]);
+    });
+
+    it('should warn if label arg is missing', function() {
+      ga.initialize('foo');
+      ga.outboundLink( { missing: 'labelarg' }, function () {
+        // do nothing on callback
+      } );
+      console.warn.args.should.eql([[
+        '[react-ga]', 'args.label is required in outboundLink()'
+      ]]);
+    });
+
+    it('should warn if hitCallback is missing', function() {
+      ga.initialize('foo');
+      ga.outboundLink( { label: 'Missing hitCallback' } );
+      console.warn.args.should.eql([[
+        '[react-ga]', 'hitCallback function is required'
+      ]]);
+    });
+
+    it('should fire hitCallback even if ga is not defined', function(done) {
+      ga.outboundLink( { label: 'http://www.google.com' }, function () {
+        done();
+      });
+    });
+
+    // TODO: solve how to simulate lack of response from GA
```

Comment:
This pushed my JS thinking a little bit further, but I got it! Thanks for your patience with this.